packages/runtime-core/__tests__/apiOptions.spec.ts

```diff
@@ -426,83 +426,117 @@ describe('api: options', () => {
 
   test('mixins', () => {
     const calls: string[] = []
-    const mixinA = {
+    const mixinA = defineComponent({
       data() {
         return {
           a: 1
         }
       },
-      created(this: any) {
+      created() {
         calls.push('mixinA created')
         expect(this.a).toBe(1)
-        expect(this.b).toBe(2)
-        expect(this.c).toBe(3)
+        // should not visit other props/data... in one mixin
+        // ????? only test current mixin props/data
+        // expect(this.b).toBe(2)
```

Comment:
This test is quite important, because is expected you to access the parent props in the mixin. 

But with typed mixins it makes quite hard to get those types using `defineComponent`(maybe another api `defineMixin<vmProps>`?). 

I think this test would be better if you remove `defineComponent` from the `mixins` and call it in the `Comp` mixin

```ts
const mixinA = {
// ...
}

const mixinB = {
//...
}

// etc

const Comp = defineComponent({
  mixins: [
    defineComponent(mixinA),
    defineComponent(mixinB),
    defineComponent(mixinC)
  ],
// etc
)
```

Or having it has two tests, one defining typed based other using non-typed (no-defineComponent)

Reply:
Very good suggestion！ 👍  
Mixin types was tested in `defineComponent-test-d.tsx` for types cases.
Here we must ensure the runtime logic is correct.